chemprop/utils/mixins.py

```diff
@@ -1,20 +1,57 @@
+"""Mixin classes for the project."""
+
 import inspect
 from typing import Any, Collection
 
 
 class ReprMixin:
     def __repr__(self) -> str:
-        items = self.get_params()
+        """Return a string representation of the object.
+
+        Returns
+        -------
+        str
+            The string representation of the object.
+        """
+        param_repr_list = []
+        for param_name, param_value in self.get_params(deep=False).items():
+            param_default = self._get_default_params()[param_name]
+            if param_value != param_default:
+                param_repr_list.append(f"{param_name}={repr(param_value)}")
+
+        non_default_params = ", ".join(param_repr_list)
+        return f"{self.__class__.__name__}({non_default_params})"
+
+    def _get_default_params(self) -> dict[str, Any]:
+        """Get the default parameters for initializing the class."""
+        sig = inspect.signature(self.__class__)
+        return {k: v.default for k, v in sig.parameters.items()}
+
+    def _get_parm_names(self) -> list[str]:
+        """Get the default parameters for initializing the class."""
```

Comment:
I don't think this docstring is correct.
```suggestion
    @classmethod
    def _get_param_names(cls) -> list[str]:
        """Get the default parameters for initializing the class."""
```

Reply:
Also, why even define this method at all? It's just syntactic sugar for `self._get_default_params().keys()`. It's not  clear to me what advantage abstracting that call into a method confers.

Reply:
It looks like c-w-feldmann was following the style of scikit-learn, which has `_get_param_names()` as a separate method for deprecation and error checking (best I can tell). We don't necessarily need those for Chemprop and could simplify this by just calling `self._get_default_params().keys()` as you suggested.

Reply:
The docstring is indeed wrong. I'll fix it.  
I used sorted, as the `keys()` method returns a `dict_keys` object and not a `list`.
(Using `list` would equally work, but I had some years ago problems with reproducing results exactly, where scripts used `sets`. So maybe I am just a bit paranoid.)
This is just nitpicking, so I am also fine with returning `self._get_default_params().keys()`